test/rekt/resources/pingsource/pingsource.go

```diff
@@ -53,28 +53,7 @@ func IsReady(name string, timings ...time.Duration) feature.StepFn {
 }
 
 // WithSink adds the sink related config to a PingSource spec.
-func WithSink(ref *duckv1.KReference, uri string) manifest.CfgFn {
-	return func(cfg map[string]interface{}) {
-		if _, set := cfg["sink"]; !set {
-			cfg["sink"] = map[string]interface{}{}
-		}
-		sink := cfg["sink"].(map[string]interface{})
-
-		if uri != "" {
-			sink["uri"] = uri
-		}
-		if ref != nil {
-			if _, set := sink["ref"]; !set {
-				sink["ref"] = map[string]interface{}{}
-			}
-			sref := sink["ref"].(map[string]interface{})
-			sref["apiVersion"] = ref.APIVersion
-			sref["kind"] = ref.Kind
-			// skip namespace
```

Comment:
Just noticed that we skip namespace here, and [apiserversource has almost the identical code](https://github.com/benmoss/eventing/blob/7dc0a77e2c47bb3a8fd9dd54fb9963b00d17215a/test/rekt/resources/apiserversource/apiserversource.go#L95) but including the namespace. Any idea on the significance of the difference?

Reply:
for the most part, all the ref's namespaces need to be in the same namespace as the referring resource. I think there are some exceptions like ... Trigger? but I would have to go look at the spec to be sure.